### ISSUE TITLE
bug(material select): Accessibility iOS VoiceOver looses focus when selecting options

### ISSUE THREAD

**Issue by yannyannyann** [angular/components#33209]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

WCAG issue. 

Issue seen with iPhone and VoiceOver on the following versions and browsers
- Safari with angular material 20 and 22
- Chrome with angular material 21 and 22

The main issue is that after a double tap on the dropdown (action to select an element when voice over is enabled), when swiping between elements, the focus continues to move outside the select, instead of browsing each element in the select.

The issue is not reproducible on Mac, nor on a classic desktop, and is restricted to VoiceOver use on an iPhone. 
The issue is not reproducible on Android/Talkback neither. 

### Reproduction

Angular Material Example page: https://material.angular.dev/components/select/overview 
StackBlitz link: https://stackblitz.com/run?file=src%2Fexample%2Fselect-reset-example.ts&startScript=start

Steps to reproduce:
1. activate VoiceOver on your iPhone/iPad device
2. Visit the angular material select menu example page 
3. play with the select menus having options list that overlap other focusable DOM elements (another menu, a heading, etc...)

### Expected Behavior

Expected: The screen reader focus should go back to the menu where the option is selected. 

### Actual Behavior

Actual: After selection is done in the menu, the screen reader focus is pointing to the element behind the chosen material-option.

### Environment

- Angular: 19, 20, 21, 22
- CDK/Material: select
- Browser(s): Safari/Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): iOS

**Comment by yannyannyann** [comment]
Demo videos of the issue

Chrome with angular material 22
https://github.com/user-attachments/assets/6d87c4ed-08a7-402b-b6b9-39951d23cdcf

Chrome with angular material 21
https://github.com/user-attachments/assets/6bed7831-d3ea-424f-9c0f-12d78d936aaf

Safari with angular material 22
https://github.com/user-attachments/assets/48f11584-6303-4443-b2d2-b62d482ca6bb

Safari with angular material 20
https://github.com/user-attachments/assets/991b190b-1818-4733-9c83-fbbe3c512883

**Comment by yannyannyann** [comment]
@crisbeto  videos were added to help you understand this issue. 
Could you please view them? thanks a lot :) 

The fix provided here does not work actually ... I will need to close the PR
https://github.com/angular/components/pull/33210

**Comment by crisbeto** [comment]
Yep, I could see it as well on my phone. I wonder if this isn't a case of us restoring focus too early.

**Comment by yannyannyann** [comment]
@crisbeto , Do you have any guidance on where to start fixing this issue? 
I have been trying several things at Mat-Select level but maybe it is to be done elsewhere?
